js/libpcre/runtime.js

```diff
@@ -372,6 +381,57 @@ function pcre_exec_stub_bc(
   }
 }
 
+//Provides: pcre_names_stub
+//Requires: libpcre, PCRE_INFO_NAMECOUNT, PCRE_INFO_NAMEENTRYSIZE, PCRE_INFO_NAMETABLE, caml_js_to_array
+function pcre_names_stub(v_rex) {
+  const { regexp_ptr, extra_ptr } = v_rex;
+
+  var name_count_ptr = libpcre._malloc(4);
+  var entry_size_ptr = libpcre._malloc(4);
+  var tbl_ptr_ptr = libpcre._malloc(4);
+
+  var ret = libpcre._pcre_fullinfo(
+    regexp_ptr,
+    extra_ptr,
+    PCRE_INFO_NAMECOUNT,
+    name_count_ptr
+  );
+  if (ret != 0) throw new Error("pcre_names_stub: namecount");
+
+  ret = libpcre._pcre_fullinfo(
+    regexp_ptr,
+    extra_ptr,
+    PCRE_INFO_NAMEENTRYSIZE,
+    entry_size_ptr
+  );
+  if (ret != 0) throw new Error("pcre_names_stub: nameentrysize");
+
+  ret = libpcre._pcre_fullinfo(
+    regexp_ptr,
+    extra_ptr,
+    PCRE_INFO_NAMETABLE,
+    tbl_ptr_ptr
+  );
+  if (ret != 0) throw new Error("pcre_names_stub: nametable");
+
+  var result = [];
+
+  const name_count = libpcre.getValue(name_count_ptr, "i32");
+  const entry_size = libpcre.getValue(entry_size_ptr, "i32");
+  var tbl_ptr = libpcre.getValue(tbl_ptr_ptr, "i32");
+
+  for (var i = 0; i < name_count; i++) {
+    result[i] = libpcre.UTF8ToString(tbl_ptr + 2);
+    tbl_ptr += entry_size;
+  }
+
+  libpcre._free(name_count_ptr);
+  libpcre._free(entry_size_ptr);
+  libpcre._free(tbl_ptr_ptr);
+
+  return caml_js_to_array(result);
```

Comment:
Unrelated to review: These 5 lines cause me so much pain. Perl common regular expressions. Pointers. Memory freeing. OCaml to JS. All in a javascript file. Compiled to WASM. Pain

Reply:
This is why they pay us the big bucks 💸